druid/src/menu.rs

```diff
@@ -120,6 +125,35 @@ pub struct MenuDesc<T> {
     items: Vec<MenuEntry<T>>,
 }
 
+/// A platform-agnostic description of an application, window, or context
+/// menu. The user has to guarantee that this represents a `MenuDesc<T>` where `T` is the users
+/// `AppState`.
+pub struct AppStateMenuDesc {
+    inner: Box<dyn Any>,
+    type_name: &'static str,
+}
+
+impl<T: Any> MenuDesc<T> {
+    /// This turns a typed `MenuDesc<T>` into an untyped `AppStateMenuDesc`.
+    /// Doing so allows sending `MenuDesc` through `Command`s.
+    /// It is up to you, to ensure that this `T` represents your application
+    /// state that you passed to `AppLauncher::launch`.
+    pub fn into_app_state_menu_desc(self) -> AppStateMenuDesc {
+        AppStateMenuDesc {
+            inner: Box::new(self),
+            type_name: any::type_name::<Self>(),
+        }
+    }
+}
+
+impl AppStateMenuDesc {
+    pub(crate) fn realize<T: Any>(&self) -> Result<&MenuDesc<T>, AppStateTypeError> {
```

Comment:
don't mind this name, but I might just call it `downcast` or `to_concrete`? (ditto the other one)

Reply:
`downcast` seams reasonable.